packages/preview/src/browser/markdown/markdown-preview-handler.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2018 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable } from "inversify";
+import { ResourceProvider } from "@theia/core";
+import { Workspace } from '@theia/languages/lib/common';
+import { PreviewHandler } from '../preview-handler';
+import { PREVIEW_WIDGET_CLASS } from '../preview-widget';
+import URI from "@theia/core/lib/common/uri";
+
+import * as hljs from 'highlight.js';
+import * as markdownit from 'markdown-it';
+
+@injectable()
+export class MarkdownPreviewHandler implements PreviewHandler {
+
+    @inject(ResourceProvider)
+    protected readonly resourceProvider: ResourceProvider;
+
+    canHandle(uri: URI): boolean {
+        return uri.path.ext === '.md';
+    }
+
+    @inject(Workspace)
+    protected readonly workspace: Workspace;
```

Comment:
it does not seem to be used

it would be nice if unused code which is not present in master won't be removed by separate commits, you can fix such changes up to the original commit

Reply:
I'd like to squash all children of commit https://github.com/theia-ide/theia/pull/1051/commits/911d7735a3df631ab3db8b6a82d3083df8d3bd87 in the end. Would that be okay?

Reply:
it is fine, i usually do interactive rebase immediately to preserve logical order of commits locally